var/spack/repos/builtin/packages/dakota/package.py

```diff
@@ -28,31 +28,46 @@ class Dakota(CMakePackage):
     homepage = 'https://dakota.sandia.gov/'
     url = 'https://dakota.sandia.gov/sites/default/files/distributions/public/dakota-6.12-release-public.src.tar.gz'
 
+    version('6.13', sha256='bca9448ebbcc5a29226eab2abbd9e5e87ccaef6818658edb64588f234bf7bd1b')
     version('6.12', sha256='4d69f9cbb0c7319384ab9df27643ff6767eb410823930b8fbd56cc9de0885bc9')
-    version('6.9', sha256='989b689278964b96496e3058b8ef5c2724d74bcd232f898fe450c51eba7fe0c2')
-    version('6.3', sha256='0fbc310105860d77bb5c96de0e8813d75441fca1a5e6dfaf732aa095c4488d52')
+    version('6.11', sha256='d38bbfccba4ff5f2187bdbb45633e269ef4d9f68631a17a52b4d05b5f9b75f65')
+    version('6.10', sha256='2a0d2f426c0f369cfa3c0b341061553e651a6f60377a59f83251d9d3c3821bad')
+    version('6.9', sha256='ede7149843707f4b07e76aae27e6a6826734131938da8a6c1b7ed11865c7ee84')
+    version('6.8', sha256='43dc6027b7f666478f2aff7663803463cef98ab2269a55f69c02ada861659c24')
+    version('6.3', sha256='0fbc310105860d77bb5c96de0e8813d75441fca1a5e6dfaf732aa095c4488d52', url='https://dakota.sandia.gov/sites/default/files/distributions/public/dakota-6.3-public.src.tar.gz')
 
     variant('shared', default=True,
             description='Enables the build of shared libraries')
     variant('mpi', default=True, description='Activates MPI support')
+    variant('hdf5', default=False, description='Activate HDF5 support')
 
     # Generic 'lapack' provider won't work, dakota searches for
     # 'LAPACKConfig.cmake' or 'lapack-config.cmake' on the path
     depends_on('netlib-lapack')
 
     depends_on('blas')
     depends_on('mpi', when='+mpi')
+    depends_on('python@3:', when='@6.8:')
+    depends_on('python@:2', when='@:6.8')
```

Comment:
Version ranges are inclusive, so this will require both Python 2 and 3 for 6.8, which isn't possible.